eggo/dag.py

```diff
@@ -69,6 +69,40 @@ class ToastConfig(Config):
     config = JsonFileParameter()
 
 
+def _cghub_download(url, tmp_dir, cghub_key=None, n_threads=8):
+    """Download from CGHub to TMP_DIR.
+
+    Requires GeneTorrent. Download client `gtdownload` must be
+    on PATH. Use public key if none provided and CGHUB_KEY
+    environment variable not set. Returns analysis subdirectory.
+    """
+    # 1. Check env for CGHub key and substitute public if necessary
+    if cghub_key is None:
+        cghub_key = os.environ.get('CGHUB_KEY') or CGHUB_PUBLIC_KEY
+
+    # 2. Parse url for analysis ID and filename
+    analysis_id, filename = url.lstrip('cghub://').split('/')
```

Comment:
yes, the CGHub metadata store centers around the `analysis_id`. It refers to a single downloadable object (which may contain multiple files). I made up a cghub url to fit the existing registry structure, so it is easy to change. Right now I am creating them to only contain the analysis ID and a single filename of interest (BAM file, generally), so this would always split into 2 objects. 

However, the CGHub REST API returns a JSON with lots of metadata for a given analysis_id. One option would be to store this full JSON in the registry, though it would be long and cluttered, not as easily human-readable. Or we could just use the analysis ID and have the code call the API to get the filename, filesize, and other metadata if necessary.